swimster-ui/src/components/App/App.jsx

```diff
@@ -7,64 +7,46 @@ import Navbar from '../Navbar/Navbar'
 import apiClient from '../../services/apiClient'
 import ProtectedRoute from '../ProtectedRoute/ProtectedRoute'
 import CreatePool from '../CreatePool/CreatePool'
+import ListingDetail from '../ListingDetail/ListingDetail'
 import { useAuthContext } from '../../contexts/auth'
+import { useListingsContext } from '../../contexts/listings'
 import { style } from './style'
 
 const App = () => {
-  const [listings, setListings] = useState([])
-  const { user, setUser } = useAuthContext()
-  const [error, setError] = useState(null)
-  const [isFetching, setIsFetching] = useState(false)
+  const { handlers: authHandlers, isUserAuthenticated, setInitialized } = useAuthContext()
+  const { handlers: listingsHandlers } = useListingsContext()
 
   useEffect(() => {
-    const fetchUser = async () => {
-      const { data, error } = await apiClient.fetchUserFromToken()
-      if (data) setUser(data.user)
-      if (error) setError(error)
-    }
-
-    const token = localStorage.getItem("swimster_session_token")
-    if (token) {
-      apiClient.setToken(token)
-      fetchUser()
-    }
-  }, [setUser])
+    const initApp = async () => {
+      console.log("Initializing App...")
+      const token = localStorage.getItem("swimster_session_token")
 
-  useEffect(() =>{
-    const fetchListings = async () => {
-      setIsFetching(true)
+      if (token) {
+        apiClient.setToken(token)
+        await authHandlers.fetchUserFromToken()
+        await listingsHandlers.fetchListings()
+      }
 
-      const { data, error } = await apiClient.fetchListings()
-      if (error) setError(error)
-      if (data?.listings) setListings(data.listings)
-
-      setIsFetching(false)
+      setInitialized(true)
```

Comment:
you already setInitialized in fetchUserFromToken?

Reply:
I did, for some reason I left it there when moving it to the `auth` context. Changed!